neqo-http3/src/connection_server.rs

```diff
@@ -87,7 +89,7 @@ impl Http3ServerHandler {
             .http_stream()
             .ok_or(Error::InvalidStreamId)?
             .send_headers(headers, conn)?;
-        self.base_handler.stream_has_pending_data(stream_id);
+        self.base_handler.stream_might_have_pending_data(stream_id);
```

Comment:
If `send_headers()` succeeds, it will always buffer something, so this call being unconditional would be fine.